src/sources/envoy_als/grpc.rs

```diff
@@ -0,0 +1,100 @@
+use crate::{
+    internal_events::{EventsReceived, StreamClosedError},
+    shutdown::ShutdownSignal,
+    SourceSender,
+};
+use envoy_proto::envoy::service::accesslog::v3::{
+    access_log_service_server::AccessLogService, stream_access_logs_message::LogEntries,
+    StreamAccessLogsMessage, StreamAccessLogsResponse,
+};
+use tonic::{Request, Response, Status, Streaming};
+use value::Value;
+use vector_common::internal_event::{CountByteSize, InternalEventHandle as _, Registered};
+use vector_core::{event::LogEvent, EstimatedJsonEncodedSizeOf};
+
+#[derive(Clone)]
+pub(super) struct Service {
+    pub events_received: Registered<EventsReceived>,
+    pub pipeline: SourceSender,
+    pub shutdown: ShutdownSignal,
+}
+
+#[tonic::async_trait]
+impl AccessLogService for Service {
+    async fn stream_access_logs(
+        &self,
+        request: Request<Streaming<StreamAccessLogsMessage>>,
+    ) -> Result<Response<StreamAccessLogsResponse>, Status> {
+        let mut in_stream = request.into_inner();
+        let mut shutdown = self.shutdown.clone();
+        let mut stream_identifier = None;
+
+        loop {
+            tokio::select! {
+                _ = &mut shutdown => break,
+                stream_msg = in_stream.message() => {
+                    match stream_msg {
+                        Ok(omsg) => {
+                            match omsg {
+                                Some(msg) => {
+                                    let mut events = vec![];
+
+                                    if let Some(logs_entries) = msg.log_entries {
+                                        match logs_entries {
+                                            LogEntries::HttpLogs(http_logs) => {
+                                                for l in http_logs.log_entry {
+                                                    let mut evt = LogEvent::default();
+                                                    evt.insert("http_log", Value::from(l));
```

Comment:
Is there a reason to put it behind an `http_log` field, as opposed to putting the contents of the `http_log` into the root of the event ? Naively it seems to me like the `http_log` field isn't adding any benefit.

Reply:
It's basically following the envoy protos https://www.envoyproxy.io/docs/envoy/v1.24.1/api-v3/service/accesslog/v3/als.proto#service-accesslog-v3-streamaccesslogsmessage (but emitting an event for each log entry)

The top level `http_log` sits beside the `identifier` field and would used used to distinguish between a future `tcp_log` log field at the same level.

Reply:
🤔  curious if @StephenWakely has thoughts on this aspect.

Reply:
I think it would be best if we had the object as the root `LogEvent` - so `LogEvent::from(Value::from(l))`. We should insert an extra field to represent the access - either "http" or "tcp". This field should go into the source metadata when LogNamespacing is implemented.

I don't believe downstream components will necessarily care whether it was a tcp or http event originally and it would be a pain for vrl to have to write things like, `if .(http | tcp).field == "wobble"` when just `if .field == "wobble"` would suffice. Any time a component needs to act differently according to http or tcp it could just check the `.access` field.